src/arithmetic.jl

```diff
@@ -34,6 +33,145 @@ function ==(a::HomogeneousPolynomial, b::HomogeneousPolynomial)
+`constant_term(a)` and `b`. If they are equal, returns `a[nz] < 0`, with `nz` the first
+non-zero coefficient beyond the constant term. This defines a total order.
+
+Refs:
+- M. Berz, AIP Conference Proceedings 177, 275 (1988); https://doi.org/10.1063/1.37800
+- M. Berz, "Automatic Differentiation as Nonarchimedean Analysis", Computer Arithmetic and
+    Enclosure Methods, (1992), Elsevier, 439-450.
+
+---
+
+    <(a::Taylor1{<:Real}, b::Taylor1{<:Real})
+    <(a::TaylorN{<:Real}, b::Taylor1{<:Real})
+
+Return `a - b < zero(b)`.
+"""
+function <(a::Taylor1{<:Real}, b::Real)
+    a0 = constant_term(a)
+    a0 != b && return a0 < b
+    nz = findfirst(a-b)
+    if nz == -1
+        return zero(a0) < zero(b)
+    else
+        return a[nz] < zero(b)
+    end
+end
+<(b::Real, a::Taylor1{<:Real}) = >(a, b)
+
+function <(a::HomogeneousPolynomial{<:Real}, b::Real)
+    ord_a = get_order(a)
+    if ord_a == 0
+        return a[1] < b
+    else
+        !iszero(b) && return zero(a[1]) < b
+        nz = findfirst(a)
+        return a[nz] < zero(b)
+    end
+end
+<(b::Real, a::HomogeneousPolynomial{<:Real}) = >(a, b)
+
+function <(a::TaylorN{<:Real}, b::Real)
+    a0 = constant_term(a)
+    a0 != b && return a0 < b
+    nz = findfirst(a-b)
+    if nz == -1
+        return zero(a0) < zero(b)
+    else
+        return a[nz] < zero(b)
+    end
+end
+<(b::Real, a::TaylorN{<:Real}) = >(a, b)
+
+
+"""
+    >(a::Taylor1{<:Real}, b::Real)
+    >(a::Taylor1{<:Real}, b::Real)
+
+Greater-than comparison of `a::Taylor1` series and `b::Real`, computed by comparing
+`constant_term(a)` and `b`. If both are equal, returns `a[nz] > 0`, with `nz` the first
+non-zero coefficient beyond the constant term. This defines a total order.
+
+Refs:
+- M. Berz, AIP Conference Proceedings 177, 275 (1988); https://doi.org/10.1063/1.37800
+- M. Berz, "Automatic Differentiation as Nonarchimedean Analysis", Computer Arithmetic and
+    Enclosure Methods, (1992), Elsevier, 439-450.
+
+---
+
+>(a::Taylor1{<:Real}, b::Taylor1{<:Real})
+>(a::TaylorN{<:Real}, b::Taylor1{<:Real})
+
+Return `a - b > zero(b)`.
+"""
+function >(a::Taylor1{<:Real}, b::Real)
+    a0 = constant_term(a)
+    a0 != b && return a0 > b
+    nz = findfirst(a-b)
+    if nz == -1
+        return zero(a0) > zero(b)
+    else
+        return a[nz] > zero(b)
+    end
+end
+>(b::Real, a::Taylor1{<:Real}) = <(a, b)
+
+function >(a::HomogeneousPolynomial{<:Real}, b::Real)
+    ord_a = get_order(a)
+    if ord_a == 0
+        return a[1] > b
+    else
+        !iszero(b) && return zero(a[1]) > b
+        nz = findfirst(a)
+        return a[nz] > zero(b)
+    end
+end
+>(b::Real, a::HomogeneousPolynomial{<:Real}) = <(a, b)
+
+function >(a::TaylorN{<:Real}, b::Real)
+    a0 = constant_term(a)
+    a0 != b && return a0 > b
+    nz = findfirst(a-b)
+    if nz == -1
+        return zero(a0) > zero(b)
+    else
+        return a[nz] > zero(b)
+    end
+end
+>(b::Real, a::TaylorN{<:Real}) = <(a, b)
+
+#
+for T in (:Taylor1, :TaylorN,)
```

Comment:
If there's metaprogramming anyway, might as well also metaprogram away the `>` and `<` repetition.

Reply:
Done! Indeed, saves a lots of repetitions.